--- a/lib/boot/pkg/ghc-prim/ghc-prim.cabal	1969-12-31 19:00:01.000000000 -0500
+++ b/lib/boot/pkg/ghc-prim/ghc-prim.cabal	2021-07-17 21:22:09.246039189 -0400
@@ -19,8 +19,8 @@
     location: https://gitlab.haskell.org/ghc/ghc.git
     subdir:   libraries/ghc-prim
 
--- custom-setup
---     setup-depends: base >= 4 && < 5, Cabal >= 1.23
+custom-setup
+    setup-depends:
 
 Library
     default-language: Haskell2010
@@ -40,6 +40,8 @@
         UnliftedFFITypes
 
     build-depends: rts == 1.0.*
+                 , base
+                 , Cabal
 
     exposed-modules:
         GHC.CString
