--- a/Setup.lhs	2021-07-06 12:02:16.079849691 -0400
+++ b/Setup.lhs	2021-07-06 12:02:28.120397571 -0400
@@ -45,7 +45,7 @@
               _ -> []
 
 myPostBuild _ flags _ lbi = do
-  let runProgram p = rawSystemProgramConf (fromFlagOrDefault normal (buildVerbosity flags))
+  let runProgram p = runDbProgram (fromFlagOrDefault normal (buildVerbosity flags))
                                           p
                                           (withPrograms lbi)
       cpp_template src dst opts = do
